dropwizard-jetty/src/main/java/io/dropwizard/jetty/BiDiGzipHandler.java

```diff
@@ -87,18 +89,33 @@ private Inflater buildInflater() {
 
     private WrappedServletRequest wrapDeflatedRequest(HttpServletRequest request) throws IOException {
         final Inflater inflater = buildInflater();
-        final InflaterInputStream input = new InflaterInputStream(request.getInputStream(), inflater, inputBufferSize) {
-            @Override
-            public void close() throws IOException {
-                super.close();
-                localInflater.set(inflater);
-            }
-        };
-        return new WrappedServletRequest(request, input);
+        try {
+            final InflaterInputStream input = new InflaterInputStream(request.getInputStream(), inflater, inputBufferSize) {
+                @Override
+                public void close() throws IOException {
+                    super.close();
+                    localInflater.set(inflater);
+                }
+            };
+            InputStream exceptionHandlingInput = new ZipExceptionHandlingInputStream(input, DEFLATE);
```

Comment:
I think we could just inline the stream to the constructor without declaring a variable.